src/cloud-api-adaptor/entrypoint.sh

```diff
@@ -1,6 +1,12 @@
 #!/bin/bash
 
 CLOUD_PROVIDER=${1:-$CLOUD_PROVIDER}
+# Enabling dynamically loaded cloud provider plugin feature, disabled by default
+CLOUD_PROVIDER_PLUGIN_ENABLE=${CLOUD_PROVIDER_PLUGIN_ENABLE:-false}
```

Comment:
Can we change the name of this config to `CLOUD_PROVIDER_EXTERNAL_PLUGINS`
and use wildcard `*` to load all the plugins or a list to load specific plugins example
`CLOUD_PROVIDER_EXTERNAL_PLUGINS="libvirt"`

Reply:
Since hash of the external plugin is also used, I'm wondering if we can use something like
CLOUD_PROVIDER_EXTERNAL_PLUGINS="plugin_name1:plugin1_hash, plugin_name2:plugin2_hash"

@mkulke do you have any suggestions on the variable naming for external plugin configs ?

Reply:
From user point view, they will only use one`CLOUD_PROVIDER` in production, so use wildcard `*`  to load all plugins or a list to load specific plugins are not a valid user case?

Reply:
@liudalibj any specific reason to restrict to only one external plugin?

Reply:
We can put multiple external plugins .so files to the ${CLOUD_PROVIDER_PLUGIN_PATH} folder on worker node, but we do NOT need load them all, we only need load the one define in peer-pods-cm `CLOUD_PROVIDER`,
Because in CAA runtime we only need one `CLOUD_PROVIDER`.

Reply:
> For single one, then it might be better to change the config vars to CLOUD_PROVIDER_EXTERNAL_PLUGIN="plugin_name" CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH="path" CLOUD_PROVIDER_EXTERNAL_PLUGIN_HASH="hash"
> 
> We don't need enable var as having a plugin name indicates you need plugin_name

Is `CLOUD_PROVIDER_EXTERNAL_PLUGIN` duplicate with `CLOUD_PROVIDER`?
surely one plugin file name `aaaa.so` can have  provider name `bbbb` be defined from code level, but we can strict ask the provider name  same as plugin file name, eg for provider named `bbbb` the plugin file MUST is `bbbb`.so.

Reply:
Good thought, we can enforce the plugin file name to be same as the provider name. Not sure if it will be restrictive from practical deployment stand point.

Also, it does make sense to have a config which enables external plugins like you had done before.
So something like CLOUD_PROVIDER_EXTERNAL_PLUGIN can be boolean (default to `false`) 
followed by CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH and CLOUD_PROVIDER_EXTERNAL_PLUGIN_HASH
Btw we can also use full plugin path as value for CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH (eg /path/plugin.so)
And then verify the plugin provider name with what is defined in CLOUD_PROVIDER ?

To summarise a working config looks like this
```
ENABLE_CLOUD_PROVIDER_EXTERNAL_PLUGIN: true
CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH: /run/cloud-api-adaptor/plugins/libvirt.so
CLOUD_PROVIDER_EXTERNAL_PLUGIN_HASH: "abc12345"
```
The code verifies the plugin provider name with CLOUD_PROVIDER configured.

Wdyt ?

Let's conclude this before you make any additional code changes :-)

Reply:
Agree, the config with full path looks good to me.
```
ENABLE_CLOUD_PROVIDER_EXTERNAL_PLUGIN: true
CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH: /run/cloud-api-adaptor/plugins/libvirt.so
CLOUD_PROVIDER_EXTERNAL_PLUGIN_HASH: "abc12345"
```
One thing need be noticed that the `/run/cloud-api-adaptor/plugins/` folder MUST have `execute` permission, otherwise when CAA load the plugin it will report "failed to map segment from shared object" error.